addons/survey/controllers/main.py

```diff
@@ -354,16 +358,31 @@ def survey_display_page(self, survey_token, answer_token, **post):
         return request.render('survey.survey_page_fill',
             self._prepare_survey_data(access_data['survey_sudo'], access_data['answer_sudo'], **post))
 
-    @http.route('/survey/get_background_image/<string:survey_token>/<string:answer_token>', type='http', auth="public", website=True, sitemap=False)
-    def survey_get_background(self, survey_token, answer_token):
+    @http.route('/survey/get_background_image/<string:survey_token>/<string:answer_token>/<int:question_id>', type='http', auth="public", website=True, sitemap=False)
+    def survey_get_background(self, survey_token, answer_token, question_id=0):
```

Comment:
Note sure, but couldn't that be inlined in _get_section_info_by_question ? Like for each question, we have its background_image_url returned and we can use it directly.